metrics/ci_integrations/lib/integration/ci/config/model/sync_config.dart

```diff
@@ -12,18 +13,32 @@ class SyncConfig extends Equatable {
   /// the loaded metrics will be saved to.
   final String destinationProjectId;
 
+  /// A number of builds to fetch from the source during project's
+  /// first synchronization.
+  final int firstSyncFetchLimit;
+
   @override
-  List<Object> get props => [sourceProjectId, destinationProjectId];
+  List<Object> get props => [
+        sourceProjectId,
+        destinationProjectId,
+        firstSyncFetchLimit,
+      ];
 
-  /// Creates an instance of this configuration.
+  /// Creates an instance of the [SyncConfig] with the given parameters.
+  ///
+  /// All parameters are required.
+  ///
+  /// Throws an [ArgumentError] if any of the given parameters is `null`.
   ///
-  /// Both [sourceProjectId] and [destinationProjectId] are required.
-  /// If one of these values is `null`, throws an [ArgumentError].
+  /// Throws an [ArgumentError] if the given [firstSyncFetchLimit] is not
+  /// greater than `0`.
   SyncConfig({
     @required this.sourceProjectId,
     @required this.destinationProjectId,
+    @required this.firstSyncFetchLimit,
   }) {
     ArgumentError.checkNotNull(sourceProjectId, 'sourceProjectId');
     ArgumentError.checkNotNull(destinationProjectId, 'destinationProjectId');
+    NumberValidator.checkGreaterThan(firstSyncFetchLimit, 0);
```

Comment:
```suggestion
    if (firstSyncFetchLimit == null || firstSyncFetchLimit <= 0) {
      throw ArgumentError(
          'The first sync fetch limit must be an integer value grater than 0');
    }
```
Would it be better since it allows us to provide a human-readable error message?